src/components/ModalCentered.tsx

```diff
@@ -0,0 +1,176 @@
+    >
+      <ModalHeader contrast={!!header}>
+        <TitleRow>
+          <PanelTitle smaller useLayoutId={false}>
+            {title}
+            {subtitle && <ModalSubtitle>{subtitle}</ModalSubtitle>}
+          </PanelTitle>
+          <CloseButton squared transparent onClick={onClose}>
+            <X />
+          </CloseButton>
+        </TitleRow>
+        {header && <ModalHeaderContent>{header}</ModalHeaderContent>}
+      </ModalHeader>
+      <ModalContent>{children}</ModalContent>
+      {isLoading && (
+        <>
+          <ModalBackdrop light />
+          <ModalLoadingSpinner>
+            <Spinner />
+          </ModalLoadingSpinner>
+        </>
+      )}
+    </StyledModal>
+  </Modal>
+)
+
+export const HeaderContent = styled(Section)`
+  flex: 0;
+  margin-bottom: var(--spacing-4);
+`
+
+export const HeaderLogo = styled.div`
+  height: 10vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+`
+
+const StyledModal = styled(motion.div)`
+  display: flex;
+  flex-direction: column;
+  margin: auto;
+  width: 100%;
+  max-width: 600px;
+  max-height: 95vh;
+  box-shadow: 0 30px 30px var(--color-shadow-15);
+  border-radius: var(--radius);
+  background-color: ${({ theme }) => theme.bg.primary};
+  z-index: 1;
+  position: relative;
+
+  ${TitleContainer} {
+    flex: 1;
+    margin: var(--spacing-3) var(--spacing-4) var(--spacing-3) var(--spacing-4);
+  }
+`
+
+export const ModalHeader = styled.header<{ contrast?: boolean }>`
+  margin-bottom: var(--spacing-3);
+
+  ${({ contrast }) =>
+    contrast &&
+    css`
+      background-color: ${({ theme }) => theme.bg.secondary};
+      border-bottom: 1px solid ${({ theme }) => theme.border.secondary};
+    `}
+`
+
+const ModalHeaderContent = styled.div`
+  display: flex;
+  justify-content: center;
+  margin-bottom: 45px;
+`
+
+const TitleRow = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const CloseButton = styled(Button)`
+  color: ${({ theme }) => theme.font.primary};
+  margin-right: var(--spacing-2);
+`
+
+const ModalContent = styled.div`
+  overflow-y: auto;
+  overflow-x: hidden;
+  padding: 0 var(--spacing-4) var(--spacing-4) var(--spacing-4);
+`
+
+export const ModalFooterButtons = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  gap: 18px;
+  padding: var(--spacing-3) var(--spacing-3) 0;
+`
+
+export const ModalFooterButton = ({ ...props }) => (
+  <ModalFooterButtonStyled short {...props}>
+    {props.children}
+  </ModalFooterButtonStyled>
+)
+
+const ModalFooterButtonStyled = styled(Button)`
+  min-width: 111px;
+  height: 30px;
+`
+
+const ModalSubtitle = styled.div`
+  color: ${({ theme }) => theme.font.secondary};
+  font-size: 14px;
+  margin-top: var(--spacing-1);
+`
+
+const ModalLoadingSpinner = styled.div`
+  position: absolute;
+  display: flex;
+  width: 100%;
+  height: 100%;
+  justify-content: center;
+  align-items: center;
+`
+
+export default ModalCentered
```

Comment:
Do we put the default export at the top of the styled component for consistency sake (as discussed above), or do we make a diff wether the container is styled or not?

Reply:
Let's keep them always at the top for consistency